test/function/target.cpp

```diff
@@ -162,6 +162,22 @@ namespace {
     EXPECT_TRUE( set_malloc_result == malloc );
   }
 
+    TEST_F( TargetTest, AppNameNotAscii ) {
+    struct stumpless_entry *bad_stump;
+    const struct stumpless_error *error;
+
+
+    const char bad_msgid[] = { 'b', 'a', 'd', '_', 0x5, 'n', 'a', 'm', 'e', '\0' };
+    bad_stump = stumpless_new_entry( STUMPLESS_FACILITY_USER,
+                                     STUMPLESS_SEVERITY_INFO,
+                                     "basic-app-name",
+                                     bad_msgid,
+                                     "basic_message" );
+
+    EXPECT_ERROR_ID_EQ( STUMPLESS_INVALID_ENCODING );
+  }
+
+
```

Comment:
This test should be removed and re-written as a non-fixture test to cover `stumpless_set_target_default_app_name` - it is currently identical to the previous test. Refer to the similar test for the default msgid which you can find in the same file at [line 639](https://github.com/goatshriek/stumpless/blob/b0e8b2ff6e7a669060147538ef16e03f76f48545/test/function/target.cpp#L639).